src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -6393,8 +6399,53 @@ void QueryAnalyzer::resolveTableFunction(QueryTreeNodePtr & table_function_node,
     QueryExpressionsAliasVisitor & expressions_visitor,
     bool nested_table_function)
 {
+
+    String database_name = scope.context->getCurrentDatabase();
+    String table_name;
+
+    if (table_function_node->getOriginalAST() && table_function_node->getOriginalAST()->as<ASTFunction>())
+    {
+        table_name = table_function_node->getOriginalAST()->as<ASTFunction>()->name;
+        if (table_function_node->getOriginalAST()->as<ASTFunction>()->is_compound_name)
+        {
+            std::vector<std::string> parts;
+            splitInto<'.'>(parts, table_function_node->getOriginalAST()->as<ASTFunction>()->name);
```

Comment:
what if table name or database name contains dots?

Reply:
This is how we separated the names here as well (https://github.com/ClickHouse/ClickHouse/blob/master/src/Interpreters/Context.cpp#L1718) , so we dont support if table name had '.'. 
Is it supported for normal table creation with '.' ?